Tools/Scripts/libraries/webkitapipy/webkitapipy/sdkdb_from_build_products.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+import os
+import shutil
+import stat
+import subprocess
+import tempfile
+from typing import cast
+from pathlib import Path
+
+
+def is_executable_file(path):
+    ext = os.path.splitext(path)[1]
+    mode = stat.S_IFREG | stat.S_IXUSR
+    return ext == '.a' or ext in ('', '.dylib') and \
+        os.stat(path).st_mode & mode == mode
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='wrapper around `tapi sdkdb` to produce a partial sdkdb '
+        'from Mach-O build products.')
+    parser.add_argument('products', nargs='+', type=Path,
+                        help='frameworks, dylibs, or root directories to '
+                        'include in generated sdkdb')
+    parser.add_argument('--sdk-dir',
+                        help='SDK to use when parsing public headers')
+    parser.add_argument('--output',
+                        help='path to write resulting .sdkdb to')
+    args = parser.parse_args()
+
+    sdk_dir = args.sdk_dir or os.environ['SDKROOT']
+    output_file = args.output or \
+        args.products[0].with_suffix('.partial.sdkdb').name
+
+    workdir = tempfile.TemporaryDirectory(prefix='sdkdb-from-build-products')
+    sdk_content = Path(workdir.name, 'SDKContentRoot')
+    runtime_content = Path(workdir.name, 'Root')
+    tapi_output = Path(workdir.name, 'sdkdb/partial.sdkdb')
+
+    for product in cast(list[Path], args.products):
+        parts = product.parts
+        for idx in reversed(range(len(parts))):
+            part = parts[idx]
+            if part.endswith('.framework'):
+                path_in_root = Path('System/Library/PrivateFrameworks',
```

Comment:
The hardcoded paths feel dangerous, but I don't have a better suggestion. Perhaps this needs some comments, with the expectation that other people will have to update this code in the future?